ckanext/switzerland/tests/test_sbb_harvester.py

```diff
@@ -93,51 +102,62 @@ def test_existing_resource_same_filename(self):
         data.resource(dataset=dataset, filename=data.filename)
 
         MockFTPStorageAdapter.filesystem = self.get_filesystem()
-        self.run_harvester()
+        self.run_harvester(ftp_server="testserver")
 
         dataset = self.get_dataset()
 
-        assert_equal(len(dataset["resources"]), 1)
+        self.assertEqual(len(dataset["resources"]), 1)
         resource = dataset["resources"][0]
 
-        assert_equal(resource["title"]["de"], "AAAResource")
-        assert_equal(resource["description"]["de"], "AAAResource Desc")
+        self.assertEqual(resource["title"]["de"], "AAAResource")
+        self.assertEqual(resource["description"]["de"], "AAAResource Desc")
 
+    @pytest.mark.usefixtures("with_plugins", "clean_db", "clean_index", "harvest_setup")
     def test_skip_already_harvested_file(self):
         """
         When modified date of file is older than the last harvester run date, the file
         should not be harvested again
         """
         MockFTPStorageAdapter.filesystem = self.get_filesystem()
-        self.run_harvester()
-        self.run_harvester()
+        self.run_harvester(ftp_server="testserver")
+        self.run_harvester(ftp_server="testserver")
 
-        assert_equal(harvester_model.HarvestSource.count(), 1)
-        assert_equal(harvester_model.HarvestJob.count(), 2)
+        self.assertEqual(harvester_model.HarvestSource.count(), 1)
+        self.assertEqual(harvester_model.HarvestJob.count(), 2)
 
         package = self.get_package()
 
-        assert_equal(len(package.resources), 1)
-        assert_equal(len(package.resources_all), 1)
+        self.assertEqual(len(package.resources), 1)
 
+    @pytest.mark.usefixtures("with_plugins", "clean_db", "clean_index", "harvest_setup")
     def test_force_all(self):
         """
         When modified date of file is older than the last harvester run date, the file
         should not be harvested again
         force_all overrides this mechanism and reharvests all files on the ftp server.
         """
         MockFTPStorageAdapter.filesystem = self.get_filesystem()
-        self.run_harvester(force_all=True)
-        self.run_harvester(force_all=True)
+        self.run_harvester(force_all=True, ftp_server="testserver")
 
-        assert_equal(harvester_model.HarvestSource.count(), 1)
-        assert_equal(harvester_model.HarvestJob.count(), 2)
+        package = self.get_package()
+        resource_id_1 = package.resources[0].id
+
+        self.run_harvester(force_all=True, ftp_server="testserver")
+        self.assertEqual(harvester_model.HarvestSource.count(), 1)
+        self.assertEqual(harvester_model.HarvestJob.count(), 2)
```

Comment:
when we only run the harvester once, shouldn't the harvest_job count be 1?
in the removed code it was run twice, now only once, but it still has two harvest jobs?

Reply:
It's still run twice, though it's a little hard to see on the diff: lines 140 and 145.

Reply:
ah, you are right, sorry :D thank you for pointing me to it 😅